txpool/txpool.go

```diff
@@ -64,6 +64,44 @@ type signer interface {
 	Sender(tx *types.Transaction) (types.Address, error)
 }
 
+// Gauge for measuring pool capacity in slots
+type slotGauge struct {
+	sync.Mutex
+	current uint64
+	limit   uint64
+}
+
+func (g *slotGauge) increase(slots uint64) {
+	g.Lock()
+	defer g.Unlock()
+
+	g.current += slots
+}
+
+func (g *slotGauge) decrease(slots uint64) {
+	g.Lock()
+	defer g.Unlock()
+
+	g.current -= slots
+}
+
+func (g *slotGauge) height() uint64 {
+	g.lock()
+	defer g.unlock()
+
+	return g.current
+}
+
+// must come before unlock()
```

Comment:
These methods are unnecessary since you embed sync.Mutex.

Reply:
Agreed.

[Fixed](https://github.com/0xPolygon/polygon-sdk/pull/213/commits/268b2c90d24759c95da79201eda539751a253842)